[flake test] GitHub rate-limiter failed when installing kustomize

#### Which jobs are flaking:
.github/workflows/installation-chart.yaml

https://github.com/karmada-io/karmada/blob/cf45b575690aaaa4c3d3b4a81ef2df43670da0a0/.github/workflows/installation-chart.yaml#L46-L49

#### Which test(s) are flaking:
See https://github.com/karmada-io/karmada/actions/runs/16334984950/job/46145193662
```
Run syntaqx/setup-kustomize@v1
  with:
    kustomize-version: 5.6.0
  env:
    HELM_VERSION: v3.17.3
    KUSTOMIZE_VERSION: 5.6.0
Run curl -O "https://raw.githubusercontent.com/kubernetes-sigs/kustomize/master/hack/install_kustomize.sh"
  % Total    % Received % Xferd  Average Speed   Time    Time     Time  Current
                                 Dload  Upload   Total   Spent    Left  Speed

  0     0    0     0    0     0      0      0 --:--:-- --:--:-- --:--:--     0
100  4590  100  4590    0     0  77521      0 --:--:-- --:--:-- --:--:-- 77796
Run bash ./install_kustomize.sh 5.6.0 /tmp
  bash ./install_kustomize.sh 5.6.0 /tmp
  shell: /usr/bin/bash --noprofile --norc -e -o pipefail {0}
  env:
    HELM_VERSION: v3.17.3
    KUSTOMIZE_VERSION: 5.6.0
Github rate-limiter failed the request. Either authenticate or wait a couple of minutes.
Error: Process completed with exit code 1.
```

#### Reason for failure:
The action `syntaqx/setup-kustomize` doesn't use a customized GitHub token to fetch the binary.

#### Anything else we need to know:
There are a lot of alternatives from the action maketplace: https://github.com/marketplace?query=kustomize&page=2

--- a/.github/workflows/installation-chart.yaml
+++ b/.github/workflows/installation-chart.yaml
@@ -47,6 +47,8 @@ jobs:
         uses: syntaqx/setup-kustomize@v1
         with:
           kustomize-version: ${{ env.KUSTOMIZE_VERSION }}
+        env:
+          GITHUB_TOKEN: ${{ github.token }}
 
       - name: Run chart-testing (template)
         run: |

